app/assets/v2/js/pages/profile-trust.js

```diff
@@ -518,6 +518,176 @@ Vue.component('poap-verify-modal', {
+                        We'll check for POH registration.
+                      </p>
+                      <div class="input-group">
+                        <input type="text" readonly class="form-control" placeholder="eth-address" aria-label="handle" aria-describedby="basic-addon1" required maxlength="255" v-model="ethAddress">
+                      </div>
+                      <div v-if="validationError !== ''" style="color: red">
+                        <small>[[validationError]]</small>
+                      </div>
+                      <b-button @click="clickedValidate" :disabled="validationStep === 'perform-validation'" variant="primary" class="btn-primary mt-3 mb-2" size="lg">
+                        <b-spinner v-if="validationStep === 'perform-validation'" type="grow"></b-spinner>
+                        Validate
+                      </b-button>
+                      <br />
+                        <a href="" @click="clickedChangeWallet" :disabled="validationStep === 'perform-validation'">
+                          Change Wallet
+                        </a>
+                      <br />
+                      <a href="" v-if="validationError !== ''" @click="clickedGoBack">
+                        Go Back
+                      </a>
+                    </div>
+                    <div v-if="validationStep === 'validation-complete'">
+                      Your POH verification was successful. Thank you for helping make Gitcoin more sybil resistant!
+                      <a href="" class="btn btn-primary px-5 mt-3 mb-2 mx-2" role="button" style="font-size: 1.3em">Done</a>
+                    </div>
+                  </div>
+                </template>
+            </b-modal>`,
+  methods: {
+    dismissVerification() {
+      // If we're only hiding the modal to allow wallet selection, don't emit this event, which
+      // would prevent it from popping up again after the user completes their selection.
+      if (!this.hideTemporarily) {
+        this.$emit('modal-dismissed');
+      }
+    },
+    clickedGoBack(event) {
+      event.preventDefault();
+      this.validationStep = 'validate-address';
+      this.ethAddress = '';
+      this.validationError = '';
+    },
+    getEthAddress() {
+      const accounts = web3.eth.getAccounts();
+
+      $.when(accounts)
+        .then((result) => {
+          const ethAddress = result[0];
+
+          this.ethAddress = ethAddress;
+          this.validationStep = 'validate-poh';
+          this.hideTemporarily = false;
+        })
+        .catch((_error) => {
+          this.validationError = 'Error getting ethereum accounts';
+          this.validationStep = 'validate-address';
+          this.hideTemporarily = false;
+        });
+    },
+    generateSignature() {
+      web3.eth.personal.sign('verify_poh_registration', this.ethAddress).then((signature) => {
+        this.signature = signature;
+        this.verifyPOH();
+      });
+    },
```

Comment:
maybe missing some error handling here if something fails like wallet disconnected or user deny the verification on metamask?

Reply:
Thank you for mentioning, I will try different scenarios and add the required error handlings.